### PR TITLE
typescript-fetch: add option for TypeScript 3.6+ compatible generation

### LINKED ISSUE #3798: [BUG] [typescript-fetch] TypeScript 3.6 breaks generated code
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] What's the version of OpenAPI Generator used?
- [ ] Have you search for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description
The 3.6 release of typescript has some [breaking changes](https://devblogs.microsoft.com/typescript/announcing-typescript-3-6/#breaking-changes) that break the generated code. Now the generated code ships with TypeScript 2.4, but some (or at least I) like to update this to the latest for any free optimizations that may come with it.

##### openapi-generator version

4.1.0

##### Suggest a fix

`GlobalFetch` was removed, so use it's namespaced version under `WindowOrWorkerGlobalScope`

### PR DESCRIPTION
**Pull request by someone1**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
The 3.6 release of typescript has some [breaking changes](https://devblogs.microsoft.com/typescript/announcing-typescript-3-6/#breaking-changes) that break the generated code. This PR updates the typescript-fetch generator to TypeScript 3.6 and replaces GlobalFetch with WindowOrWorkerGlobalScope.

I'd be happy to switch down to TypeScript 3.0 (I think this is the minimum version required for this change).

fixes #3798 

cc:
@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07)

### CODE REVIEW

**Comment by macjohnny** [comment]
Maybe we could introduce an option to chose typescript3+? This way it would not be a breaking change

**Comment by someone1** [comment]
@macjohnny : Any guidance on how I should add options to the client generator?

**Comment by macjohnny** [comment]
See e.g. https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java#L89

**Comment by someone1** [comment]
@macjohnny I think it's ready for review - hopefully I did this all correctly!

**Comment by macjohnny** [comment]
@someone1 thanks for your PR!
The CI reports
ERROR: Failed to run ./bin/typescript-fetch-petstore-all.sh
Exited with code 1

Can you fix this?

File: bin/typescript-fetch-petstore-typescript-three-plus.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
```
Comment (macjohnny) [inline bin/typescript-fetch-petstore-typescript-three-plus.sh]: Please add a windows-version in /bin/windows, too

Comment (macjohnny) [reply bin/typescript-fetch-petstore-typescript-three-plus.sh]: Maybe this script still needs to be made executable, i.e.

chmod u+x bin/typescript-fetch-petetore-typescript-three-plus.sh

Comment (someone1) [reply bin/typescript-fetch-petstore-typescript-three-plus.sh]: The command was `git update-index --chmod=+x bin/typescript-fetch-petstore-typescript-three-plus.sh` for me (using WSL/Windows so maybe that's what screwed this up)

**Comment by someone1** [comment]
@macjohnny - ready for review again, added the windows batch file

**Comment by macjohnny** [comment]
please remove the newline at the end of `docs/generators/typescript-fetch.md`

**Comment by someone1** [comment]
@macjohnny - done

**Pull request merged**